src/utils/admin-utils.ts

```diff
@@ -0,0 +1,26 @@
+import { FeatureCollection } from 'geojson';
+
+import {
+  getBoundaryLayerSingleton,
+  getDisplayBoundaryLayers,
+} from '../config/utils';
+
+export function getAdminLayerURL(adminLevel = 1): string {
+  const lyrs = getDisplayBoundaryLayers();
+  const adminLayer = lyrs.find(
+    lyr => lyr.adminLevelNames.length === adminLevel,
+  );
+  return adminLayer ? adminLayer.path : lyrs[0].path;
+}
+
+export function getAdminNameProperty(adminLevel: number = 1): string {
+  const { adminLevelNames } = getBoundaryLayerSingleton();
+  return adminLevelNames[adminLevel - 1];
+}
+
+export function fetchAdminLayerGeoJSON(
```

Comment:
I think we can probably simplify this logic a lot. Why do we need to re-fetch the layers?

Most likely the layers json are already accessible.

We also need to consider the case where we only have one file and the adminLevelName is what decides how we want to group the results.

Thinking about it a bit more, we could probably always run the analyis on the getBoundaryLayerSingleton() and then use the adminBoundary level after the fact to group results.

Reply:
fixed with https://github.com/WFP-VAM/prism-frontend/pull/431/commits/c6900a19270979a57f4943f39bc4fec3ad6c87d9

Now, basically pulling the admin boundary data for the zones from state.